geomstats/geometry/lower_triangular_matrices.py

```diff
@@ -15,11 +15,16 @@ class LowerTriangularMatrices(VectorSpace):
     ----------
     n : int
         Integer representing the shapes of the matrices: n x n.
+    strictly_lower : bool
```

Comment:
What about creating a new class called `StrictlyLowerTriangularMatrices`?

Even if it shares a lot of commonalities with `LowerTriangularMatrices`, mathematically (and programmatically) it looks like a new space (you notice this, as it requires to add ifs in a lot of parts in the code).